src/middleware/express-error.js

```diff
@@ -14,6 +16,29 @@ const expressError = ({errorCodes, entry, monitor}) => {
   return (err, req, res, next) => {
 
     const incidentId = uuid.v4();
+    if (!(err instanceof ErrorReply)) {
+      res.reportError(
+        'InternalServerError',
+        'Internal Server Error, incidentId: ' + incidentId,
+        {incidentId}
+      );
```

Comment:
Calling reportError will just throw another error, which is not what we want.  Instead, we need to replace `err` here with an ErrorReply containing `InternalServerError` and an incidentId and so on.

Reply:
`reportError` is converting this into an ErrorReply. if we are passing ErrorReply from here, we might have to remove ErrorReply from `reportError`

Reply:
I'm not sure what you mean.  It is converting it into an ErrorReply, but then throwing it.

Reply:
reportError is doing something like this

```
res.reportError = (code, message, details = {}) => {
      throw new ErrorReply({code, message, details});
    };
```

Reply:
I read your comment and again, I think I got it wrong previously. 
Just to be clear, you meant to change this to something like this 
```
err = new ErrorReply({
        code: 'InternalServerError',
        message: 'Internal Server Error, incidentId: ' + incidentId,
        details: {incidentId}
      });
```
Correct??